pkg/pool-linear/contracts/LinearPool.sol

```diff
@@ -664,21 +658,35 @@ abstract contract LinearPool is ILinearPool, IGeneralPool, IRateProvider, BasePo
         (uint256 lowerTarget, uint256 upperTarget) = getTargets();
         _require(_isMainBalanceWithinTargets(lowerTarget, upperTarget), Errors.OUT_OF_TARGET_RANGE);
 
-        super.setSwapFeePercentage(swapFeePercentage);
+        _setSwapFeePercentage(swapFeePercentage);
     }
 
-    function _isMainBalanceWithinTargets(uint256 lowerTarget, uint256 upperTarget) private view returns (bool) {
-        (uint256 cash, uint256 managed, , ) = getVault().getPoolTokenInfo(getPoolId(), _mainToken);
+    /**
+     * @dev Validate the swap fee, update storage, and emit an event.
+     */
+    function _setSwapFeePercentage(uint256 swapFeePercentage) internal {
+        _require(swapFeePercentage >= _getMinSwapFeePercentage(), Errors.MIN_SWAP_FEE_PERCENTAGE);
+        _require(swapFeePercentage <= _getMaxSwapFeePercentage(), Errors.MAX_SWAP_FEE_PERCENTAGE);
 
-        uint256 mainTokenBalance = _upscale(cash + managed, _scalingFactor(_mainToken));
+        _poolState = _poolState.insertUint(
+            swapFeePercentage,
+            _SWAP_FEE_PERCENTAGE_OFFSET,
+            _SWAP_FEE_PERCENTAGE_BIT_LENGTH
+        );
 
-        return mainTokenBalance >= lowerTarget && mainTokenBalance <= upperTarget;
+        emit SwapFeePercentageChanged(swapFeePercentage);
     }
 
-    function _isOwnerOnlyAction(bytes32 actionId) internal view virtual override returns (bool) {
-        return actionId == getActionId(this.setTargets.selector) || super._isOwnerOnlyAction(actionId);
+    function _getMinSwapFeePercentage() internal pure virtual returns (uint256) {
+        return _MIN_SWAP_FEE_PERCENTAGE;
+    }
+
+    function _getMaxSwapFeePercentage() internal pure virtual returns (uint256) {
+        return _MAX_SWAP_FEE_PERCENTAGE;
     }
```

Comment:
I'd just inline these constants and remove this functions

Reply:
I went back and forth on that.

Reply:
I doubt that will be required, but if so we can always add it later on.